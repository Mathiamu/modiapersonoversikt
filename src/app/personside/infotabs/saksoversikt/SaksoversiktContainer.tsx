import * as React from 'react';
import { RestReducer } from '../../../../redux/restReducers/restReducer';
import { Sakstema, SakstemaWrapper } from '../../../../models/saksoversikt/sakstema';
import styled from 'styled-components';
import theme from '../../../../styles/personOversiktTheme';
import { Innholdstittel } from 'nav-frontend-typografi';
import { AppState } from '../../../../redux/reducers';
import { Action, Dispatch } from 'redux';
import { hentSaksoversikt, reloadSaksoversikt } from '../../../../redux/restReducers/saksoversikt';
import Innholdslaster from '../../../../components/Innholdslaster';
import { STATUS } from '../../../../redux/restReducers/utils';
import SakstemaVisning from './SakstemaVisning';
import { BaseUrlsResponse } from '../../../../models/baseurls';
import { hentBaseUrls } from '../../../../redux/restReducers/baseurls';
import DokumenterVisning from './DokumenterVisning';
import LenkepanelPersonoversikt from '../../../../utils/LenkepanelPersonoversikt';
import { lenkeNorg2Frontend } from './norgLenke';
import { Person, PersonRespons } from '../../../../models/person/person';
import { Dokument, DokumentMetadata } from '../../../../models/saksoversikt/dokumentmetadata';
import DokumentOgVedlegg from './DokumentOgVedlegg';
import { connect } from 'react-redux';
import { settValgtEnkeltdokument, settValgtSakstema } from '../../../../redux/saksoversikt/saksoversiktStateReducer';
import { hentAllPersonData } from '../../../../redux/restReducers/personinformasjon';

export interface AvsenderFilter {
    fraBruker: boolean;
    fraNav: boolean;
    fraAndre: boolean;
}

interface State {
    avsenderfilter: AvsenderFilter;
}

interface StateProps {
    baseUrlReducer: RestReducer<BaseUrlsResponse>;
    saksoversiktReducer: RestReducer<SakstemaWrapper>;
    personReducer: RestReducer<PersonRespons>;
    person: Person;
    visDokument: boolean;
    valgtDokument?: DokumentMetadata;
    valgtEnkeltdokument?: Dokument;
    valgtSakstema?: Sakstema;
}

interface DispatchProps {
    hentBaseUrls: () => void;
    hentSaksoversikt: (fødselsnummer: string) => void;
    reloadSaksoversikt: (fødselsnummer: string) => void;
    setEnkeltdokument: (enkeltdokument: Dokument) => void;
    setSakstema: (sakstema: Sakstema) => void;
    hentPerson: (fødselsnummer: string) => void;
}

interface OwnProps {
    fødselsnummer: string;
}

type Props = StateProps & DispatchProps & OwnProps;

export const saksoversiktMediaTreshold = '80rem';

const SaksoversiktArticle = styled.article`
  display: flex;
  align-items: flex-start;
  @media(max-width: ${saksoversiktMediaTreshold}) {
    display: block;
  }
  .visually-hidden {
    ${theme.visuallyHidden}
  }
  > * {
    margin-bottom: ${theme.margin.layout};
  }
`;

const SakstemaListe = styled.section`
  border-radius: ${theme.borderRadius.layout};
  background-color: white;
  min-width: 24rem;
  flex-basis: 24rem;
`;

const DokumentListe = styled.section`
  border-radius: ${theme.borderRadius.layout};
  background-color: white;
  position: relative;
  flex-grow: 1;
  @media not all and (max-width: ${saksoversiktMediaTreshold}) {
      margin-left: ${theme.margin.layout};
  }
`;

class SaksoversiktContainer extends React.Component<Props, State> {

    private dokumentListeRef = React.createRef<HTMLHeadingElement>();

    constructor(props: Props) {
        super(props);
        this.state = {
            avsenderfilter: {
                fraBruker: true,
                fraNav: true,
                fraAndre: true
            }
        };
        this.oppdaterSakstema = this.oppdaterSakstema.bind(this);
        this.toggleAvsenderFilter = this.toggleAvsenderFilter.bind(this);
    }

    componentDidMount() {
        if (this.props.baseUrlReducer.status === STATUS.NOT_STARTED) {
            this.props.hentBaseUrls();
        }
        if (this.props.saksoversiktReducer.status === STATUS.NOT_STARTED) {
            this.props.hentSaksoversikt(this.props.fødselsnummer);
        }
        if (this.props.personReducer.status === STATUS.NOT_STARTED) {
            this.props.hentPerson(this.props.fødselsnummer);
        }
    }

    oppdaterSakstema(sakstema: Sakstema) {
        this.props.setSakstema(sakstema);
        if (this.dokumentListeRef.current) {
            this.dokumentListeRef.current.focus();
        }
    }

    toggleAvsenderFilter(key: keyof AvsenderFilter) {
        this.setState({
            avsenderfilter: {
                ...this.state.avsenderfilter,
                [key]: !this.state.avsenderfilter[key]
            }
        });
    }

    render() {
        const norgUrl = this.props.baseUrlReducer.status === STATUS.OK
            ? lenkeNorg2Frontend(
                this.props.baseUrlReducer.data,
                this.props.person.geografiskTilknytning,
                this.props.valgtSakstema)
            : '';

        if (this.props.visDokument && this.props.valgtDokument) {
            return (
                <DokumentOgVedlegg
                    harTilgang={true}
                    onChange={this.props.setEnkeltdokument}
                />
            );
        } else {
            return (
                <SaksoversiktArticle>
                    <Innholdstittel className="visually-hidden">Brukerens saker</Innholdstittel>
                    <Innholdslaster avhengigheter={[this.props.saksoversiktReducer, this.props.baseUrlReducer]}>
                        <div>
                            <LenkepanelPersonoversikt
                                url={norgUrl}
                            >
                                Oversikt over enheter og tema
                            </LenkepanelPersonoversikt>
                            <SakstemaListe>
                                <SakstemaVisning
                                    sakstema={this.props.saksoversiktReducer.data.resultat}
                                    oppdaterSakstema={this.oppdaterSakstema}
                                />
                            </SakstemaListe>
                        </div>
                        <DokumentListe>
                            <span ref={this.dokumentListeRef} tabIndex={-1}/>
                            <DokumenterVisning
                                sakstema={this.props.valgtSakstema}
                                avsenderFilter={this.state.avsenderfilter}
                                toggleFilter={this.toggleAvsenderFilter}
                            />
                        </DokumentListe>
                    </Innholdslaster>
                </SaksoversiktArticle>
            );
        }
    }
}

function mapStateToProps(state: AppState): StateProps {
    return ({
        baseUrlReducer: state.restEndepunkter.baseUrlReducer,
        saksoversiktReducer: state.restEndepunkter.saksoversiktReducer,
        personReducer: state.restEndepunkter.personinformasjon,
        person: state.restEndepunkter.personinformasjon.data as Person,
        visDokument: state.saksoversikt.visDokument,
        valgtDokument: state.saksoversikt.valgtDokument,
        valgtEnkeltdokument: state.saksoversikt.valgtEnkeltdokument,
        valgtSakstema: state.saksoversikt.valgtSakstema
    });
}

function mapDispatchToProps(dispatch: Dispatch<Action>): DispatchProps {
    return {
        hentBaseUrls: () =>
            dispatch(hentBaseUrls()),
        hentSaksoversikt: (fødselsnummer: string) =>
            dispatch(hentSaksoversikt(fødselsnummer)),
        reloadSaksoversikt: (fødselsnummer: string) =>
            dispatch(reloadSaksoversikt(fødselsnummer)),
        setEnkeltdokument: (enkeltdokument: Dokument) =>
            dispatch(settValgtEnkeltdokument(enkeltdokument)),
        hentPerson: fødselsnummer =>
            hentAllPersonData(dispatch, fødselsnummer),
        setSakstema: (sakstema: Sakstema) =>
            dispatch(settValgtSakstema(sakstema))
    };
}

export default connect(mapStateToProps, mapDispatchToProps)(SaksoversiktContainer);
