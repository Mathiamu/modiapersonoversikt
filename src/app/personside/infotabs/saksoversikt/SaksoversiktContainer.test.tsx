import * as React from 'react';
import * as renderer from 'react-test-renderer';
import TestProvider from '../../../../test/Testprovider';
import SaksoversiktContainer from './SaksoversiktContainer';
import SakstemaContextProvider from './SakstemaContext';

test('Viser saksoversiktcontainer med alt innhold', () => {
    const container = renderer.create(
        <TestProvider>
            <SakstemaContextProvider>
                <SaksoversiktContainer />
            </SakstemaContextProvider>
        </TestProvider>
    );

    const json = container.toJSON();
    expect(json).toMatchSnapshot();
});
