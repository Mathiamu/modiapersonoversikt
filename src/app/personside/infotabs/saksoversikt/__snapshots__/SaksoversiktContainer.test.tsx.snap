// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Viser saksoversiktcontainer med alt innhold 1`] = `
.c14 {
  text-align: center;
}

.c16 {
  text-transform: uppercase;
}

.c15 {
  font-weight: bold;
}

.c11 {
  border: none;
  padding: 0;
  border-radius: .35rem;
  cursor: pointer;
  background-color: transparent;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #0067c5;
}

.c11:focus {
  outline: none;
  box-shadow: 0 0 0 0.1875rem #FFBD66;
}

.c11:hover {
  opacity: 0.8;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c19 {
  color: #78706a;
  margin-top: 3px;
}

.c19 .typo-etikett-liten {
  line-height: 1rem;
}

.c7 {
  padding: 1.25rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c7:hover {
  box-shadow: inset 0 0 0 0.05rem #0067c5;
}

.c7:hover > button:last-child {
  padding-left: .5rem;
}

.c7 > *:first-child {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c4 {
  padding: 1.25rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  background-color: #cce1f3;
}

.c4:hover {
  box-shadow: inset 0 0 0 0.05rem #0067c5;
}

.c4:hover > button:last-child {
  padding-left: .5rem;
}

.c4 > *:first-child {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c6 {
  border: none;
  padding: 0;
  height: 2rem;
  width: 2rem;
  border-radius: .35rem;
  cursor: pointer;
  background-color: transparent;
  -webkit-transition: padding-left .3s;
  transition: padding-left .3s;
}

.c6:focus {
  outline: none;
  box-shadow: 0 0 0 0.1875rem #FFBD66;
}

.c5 svg {
  height: 1.875rem;
  width: 1.875rem;
}

.c3 > * {
  border-top: solid 0.0625rem #b7b1a9;
}

.c1 {
  border-radius: .25rem;
  background-color: white;
  min-width: 24rem;
  -webkit-flex-basis: 24rem;
  -ms-flex-preferred-size: 24rem;
  flex-basis: 24rem;
}

.c1 ol {
  list-style: none;
}

.c1 table {
  width: 100%;
  text-align: right;
  border-spacing: 0;
}

.c1 table * {
  padding: 10px;
  margin: 0;
}

.c1 table tr > * {
  padding: 0.1rem;
  text-align: right;
}

.c1 table tr > *:first-child {
  text-align: left;
}

.c1 table tr > *:not(:first-child) {
  width: 6rem;
}

.c2 {
  padding: 1.25rem;
}

.c17 {
  padding: 1.25rem 0.625rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: pointer;
}

.c17 svg {
  padding-right: 1rem;
  height: 1.875rem;
  width: auto;
}

.c17:hover {
  box-shadow: inset 0 0 0 0.05rem #0067c5;
}

.c17:active {
  background-color: rgba(0,0,0,0.1);
}

.c18 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c18 > *:not(:last-child) {
  margin-bottom: 0.5rem;
}

.c8 {
  position: relative;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c9 {
  border-radius: 0.25rem;
  background-color: white;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 1.25rem;
}

.c9 > *:first-child {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c9 > div {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c9 > div > * {
  margin-bottom: .5rem;
}

.c9 > div:last-child {
  -webkit-align-content: flex-end;
  -ms-flex-line-pack: end;
  align-content: flex-end;
}

.c12 {
  padding: 0;
  margin: 0;
  list-style: none;
}

.c13 > *:first-child {
  padding: .2rem 0.625rem;
}

.c13 ol {
  border-radius: 0.25rem;
  background-color: white;
  padding: 0;
  margin: 0;
}

.c13 ol > *:not(:first-child) {
  border-top: solid 0.0625rem #b7b1a9;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c10 > *:not(:last-child) {
  padding-right: 1rem;
}

.c10 > * {
  margin-bottom: 0;
}

.c20 {
  margin-top: 2rem;
}

.c0 .visually-hidden {
  position: absolute !important;
  height: 1px;
  width: 1px;
  overflow: hidden;
  -webkit-clip: rect(1px 1px 1px 1px);
  clip: rect(1px 1px 1px 1px);
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
}

.c0 > * {
  margin-bottom: .8rem;
}

@media (min-width:80rem) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-align-items: flex-start;
    -webkit-box-align: flex-start;
    -ms-flex-align: flex-start;
    align-items: flex-start;
  }

  .c0 > *:last-child {
    margin-left: .8rem;
  }
}

<article
  className="c0"
>
  <h1
    className="typo-innholdstittel visually-hidden"
  >
    Brukerens saker
  </h1>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <h2
        className="typo-undertittel"
      >
        Tema
      </h2>
    </div>
    <ol
      className="c3"
    >
      <li>
        <div
          className="c4"
          onClick={[Function]}
        >
          <div>
            <p
              className="typo-normal"
            >
              08.02.2018
            </p>
            <p
              className="typo-element"
            >
              Alle tema
            </p>
          </div>
          <span
            className="c5"
          />
          <button
            aria-label="Vis Alle tema"
            aria-selected={true}
            className="c6"
            onClick={[Function]}
          >
            <i
              className="nav-frontend-chevron chevronboks chevron--stor chevron--hoyre"
            />
          </button>
        </div>
      </li>
      <li>
        <div
          className="c7"
          onClick={[Function]}
        >
          <div>
            <p
              className="typo-normal"
            >
              08.02.2018
            </p>
            <p
              className="typo-element"
            >
              Sykepenger
            </p>
            <p
              className="typo-normal"
            >
              1
               
              søknad
               er 
              under behandling
              .
            </p>
          </div>
          <span
            className="c5"
          >
            <svg
              contentScriptType="text/ecmascript"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                fill="#C6C2BF"
              >
                <path
                  d="M17.5 10c1.3 0 2.5.3 3.5.9V4.5c0-.3-.2-.5-.5-.5H19V2.5c0-.3-.2-.5-.5-.5H6V.5c0-.3-.2-.5-.5-.5h-5C.2 0 0 .2 0 .5V16c0 1.1.9 2 2 2h8v-.5c0-4.1 3.4-7.5 7.5-7.5zM3 4.5V16a1 1 0 0 1-2 0V1h4v1.5c0 .3.2.5.5.5H18v1H3.5c-.3 0-.5.2-.5.5z"
                />
                <path
                  d="M17.5 11a6.5 6.5 0 0 0-5 10.7l9.2-9.1a6.5 6.5 0 0 0-4.2-1.6zM22.4 13.3l-9.1 9.1c1.1 1 2.6 1.6 4.2 1.6a6.5 6.5 0 0 0 5-10.7z"
                />
              </g>
            </svg>
          </span>
          <button
            aria-label="Vis Sykepenger"
            aria-selected={false}
            className="c6"
            onClick={[Function]}
          >
            <i
              className="nav-frontend-chevron chevronboks chevron--stor chevron--hoyre"
            />
          </button>
        </div>
      </li>
      <li>
        <div
          className="c7"
          onClick={[Function]}
        >
          <div>
            <p
              className="typo-normal"
            >
              21.11.2017
            </p>
            <p
              className="typo-element"
            >
              Tilleggsstønad arbeidssøkere
            </p>
            <p
              className="typo-normal"
            >
              1
               
              søknad
               er 
              under behandling
              .
            </p>
          </div>
          <span
            className="c5"
          >
            <svg
              contentScriptType="text/ecmascript"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                fill="#C6C2BF"
              >
                <path
                  d="M17.5 10c1.3 0 2.5.3 3.5.9V4.5c0-.3-.2-.5-.5-.5H19V2.5c0-.3-.2-.5-.5-.5H6V.5c0-.3-.2-.5-.5-.5h-5C.2 0 0 .2 0 .5V16c0 1.1.9 2 2 2h8v-.5c0-4.1 3.4-7.5 7.5-7.5zM3 4.5V16a1 1 0 0 1-2 0V1h4v1.5c0 .3.2.5.5.5H18v1H3.5c-.3 0-.5.2-.5.5z"
                />
                <path
                  d="M17.5 11a6.5 6.5 0 0 0-5 10.7l9.2-9.1a6.5 6.5 0 0 0-4.2-1.6zM22.4 13.3l-9.1 9.1c1.1 1 2.6 1.6 4.2 1.6a6.5 6.5 0 0 0 5-10.7z"
                />
              </g>
            </svg>
          </span>
          <button
            aria-label="Vis Tilleggsstønad arbeidssøkere"
            aria-selected={false}
            className="c6"
            onClick={[Function]}
          >
            <i
              className="nav-frontend-chevron chevronboks chevron--stor chevron--hoyre"
            />
          </button>
        </div>
      </li>
    </ol>
  </div>
  <section
    className="c8"
  >
    <section
      className="c9"
    >
      <div>
        <h2
          className="typo-undertittel"
        >
          Alle tema
        </h2>
        <form
          className="c10"
        >
          <div
            className="skjemaelement skjemaelement--horisontal"
          >
            <input
              checked={true}
              className="skjemaelement__input checkboks"
              id="Helt tilfeldig ID"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              className="skjemaelement__label"
              htmlFor="Helt tilfeldig ID"
            >
              Bruker
            </label>
            <div
              aria-live="assertive"
              role="alert"
            />
          </div>
          <div
            className="skjemaelement skjemaelement--horisontal"
          >
            <input
              checked={true}
              className="skjemaelement__input checkboks"
              id="Helt tilfeldig ID"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              className="skjemaelement__label"
              htmlFor="Helt tilfeldig ID"
            >
              NAV
            </label>
            <div
              aria-live="assertive"
              role="alert"
            />
          </div>
          <div
            className="skjemaelement skjemaelement--horisontal"
          >
            <input
              checked={true}
              className="skjemaelement__input checkboks"
              id="Helt tilfeldig ID"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              className="skjemaelement__label"
              htmlFor="Helt tilfeldig ID"
            >
              Andre
            </label>
            <div
              aria-live="assertive"
              role="alert"
            />
          </div>
        </form>
      </div>
      <div>
        <a
          className="lenke"
          href="https://norg2-frontend.nais.preprod.local/#/startsok?tema=SYK&gt=0118"
          rel="noopener noreferrer"
          target="_blank"
        >
          Oversikt over enheter og tema
        </a>
        <button
          className="c11"
          onClick={[Function]}
        >
          Viktig å vite om 
          Alle tema
        </button>
      </div>
    </section>
    <article
      className=""
    >
      <ol
        className="c12"
      >
        <li
          className="c13"
        >
          <h3
            className="typo-normal"
          >
            <div
              className="c14"
            >
              <span
                className="c15"
              >
                <span
                  className="c16"
                >
                  2018
                </span>
              </span>
            </div>
          </h3>
          <ol>
            <div
              className=""
              onClick={[Function]}
            >
              <div
                className="c17"
              >
                <svg
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M20.5 23.5h-17V.5h11l6 6zm-6-23v6h6m-13 1H12m-4.5 3h9m-9 3h9m-9 3h9m-9 3h9"
                    fill="none"
                    stroke="#C6C2BF"
                    strokeLinecap="round"
                    strokeLinejoin="round"
                    strokeMiterlimit="10"
                  />
                </svg>
                <div
                  className="c18"
                >
                  <p
                    className="typo-normal"
                  >
                    08.02.2018 / Fra Marie
                  </p>
                  <div>
                    <button
                      aria-label="Vis hoveddokument - Vurdering feilutbetaling/revurdering"
                      className="c11"
                      onClick={[Function]}
                    >
                      <p
                        className="typo-element"
                      >
                        Vurdering feilutbetaling/revurdering
                      </p>
                    </button>
                  </div>
                  
                  <div
                    className="c19"
                  >
                    <p
                      className="typo-etikett-liten"
                    >
                      Saksid: 
                      yg0ll4se
                       / Sakstema: 
                      Sykepenger
                    </p>
                  </div>
                </div>
                <span>
                  <a
                    className="lenke"
                    href="/modiapersonoversikt/saker/10108000398?sakstemaKode=ALLE&journalpostId=3fmounyc&dokumentId=mlub095n"
                    target="_blank"
                  >
                    Åpne i eget vindu
                  </a>
                </span>
              </div>
            </div>
          </ol>
        </li>
        <li
          className="c13"
        >
          <h3
            className="typo-normal"
          >
            <div
              className="c14"
            >
              <span
                className="c15"
              >
                <span
                  className="c16"
                >
                  2017
                </span>
              </span>
            </div>
          </h3>
          <ol>
            <div
              className=""
              onClick={[Function]}
            >
              <div
                className="c17"
              >
                <svg
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g
                    fill="#C6C2BF"
                  >
                    <path
                      d="M17 10V5.5l-.1-.4-5-5-.4-.1H.5C.2 0 0 .2 0 .5v21c0 .3.2.5.5.5h11A7.5 7.5 0 0 1 17 10zM11.5.5l5 5h-5v-5z"
                    />
                    <path
                      d="M17.5 11a6.5 6.5 0 0 0-5 10.7l9.2-9.1a6.5 6.5 0 0 0-4.2-1.6zM22.4 13.3l-9.1 9.1c1.1 1 2.6 1.6 4.2 1.6a6.5 6.5 0 0 0 5-10.7z"
                    />
                  </g>
                </svg>
                <div
                  className="c18"
                >
                  <p
                    className="typo-normal"
                  >
                    21.11.2017 / Samtalereferat
                  </p>
                  <div>
                    <button
                      aria-label="Vis hoveddokument - Spørsmål via nav.no"
                      className="c11"
                      onClick={[Function]}
                    >
                      <p
                        className="typo-element"
                      >
                        Spørsmål via nav.no
                      </p>
                    </button>
                  </div>
                  
                  <div
                    className="c19"
                  >
                    <p
                      className="typo-etikett-liten"
                    >
                      Saksid: 
                      5ptdxt2v
                       / Sakstema: 
                      Sykepenger
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </ol>
        </li>
      </ol>
      <div
        className="c20"
      />
      <div
        aria-live={undefined}
        className="alertstripe alertstripe--info"
      >
        <span
          aria-label="info"
          className="alertstripe__ikon"
        >
          <svg
            focusable="false"
            height="1.5em"
            kind="info-sirkel-orange"
            onClick={undefined}
            style={undefined}
            viewBox="0 0 24 24"
            width="1.5em"
          >
            <title>
              Informasjon
            </title>
            <g
              fill="#D87F0A"
              fillRule="evenodd"
            >
              <path
                d="M12 0C5.383 0 0 5.383 0 12s5.383 12 12 12 12-5.383 12-12S18.617 0 12 0zm0 22.957C5.958 22.957 1.043 18.042 1.043 12 1.043 5.958 5.958 1.043 12 1.043c6.042 0 10.957 4.915 10.957 10.957 0 6.042-4.915 10.957-10.957 10.957z"
              />
              <path
                d="M15.13 18.783h-2.608v-8.87A.522.522 0 0 0 12 9.391H9.913a.522.522 0 0 0 0 1.044h1.565v8.348H8.87a.522.522 0 0 0 0 1.043h6.26a.522.522 0 0 0 0-1.043z"
              />
              <ellipse
                cx="11.478"
                cy="5.739"
                rx="1.043"
                ry="1.043"
              />
            </g>
          </svg>
        </span>
        <span
          className="typo-normal alertstripe__tekst"
        >
          Modia viser elektroniske dokumenter brukeren har sendt inn via nav.no etter 9. desember 2014. Dokumenter som er journalført vises fra og med 4.juni 2016
        </span>
      </div>
    </article>
  </section>
</article>
`;
